nmmo/task/predicate/gold_predicate.py

```diff
@@ -0,0 +1,42 @@
+#pylint: disable=invalid-name, unused-argument
+from nmmo.task.predicate.core import predicate
+from nmmo.task.group import Group
+from nmmo.task.game_state import GameState
+
+@predicate
+def HoardGold(gs: GameState,
+              subject: Group,
+              amount: int):
+  """True iff the summed gold of all teammate is greater than or equal to amount.
+  """
+  return subject.gold.sum() / amount
+
+#######################################
+# Event-log based predicates
+#######################################
+
+@predicate
+def EarnGold(gs: GameState,
+             subject: Group,
+             amount: int):
+  """ True if the total amount of gold earned is greater than or equal to amount.
+  """
+  return subject.event.EARN_GOLD.gold.sum() / amount
```

Comment:
TODO: The list of events should be documented.

Reply:
Give this to me so I can give it to Rose